src/thread.cr

```diff
@@ -6,92 +6,125 @@ class Thread
   # Don't use this class, it is used internally by the event scheduler.
   # Use spawn and channels instead.
 
-  @th : LibC::PthreadT?
+  # all thread objects, so the GC can see them (it doesn't scan thread locals)
```

Comment:
boehm does support thread local storage. I thought the original set was added because openbsd didn't have thread local support.

Reply:
I reworded the doc, but yes, that's what it said. Maybe we're wrong, yet there's nothing bad to keep the list of threads anyway :)